portfolio/src/main/java/com/google/sps/data/CommentsDatastore.java

```diff
@@ -0,0 +1,85 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.ImmutableList; 
+
+/** Class that handles writing/reading data from the datastore. */
+public final class CommentsDatastore {
+
+  /** Class to create Comment object. */
+  public final static class Comment {
+    private final String comment;
+    private final long timestamp;
+
+    public Comment(String comment, long timestamp) {
+      this.comment = comment;
+      this.timestamp = timestamp;
+    }
+
+    /** Returns comment. */
+    public String getComment() {
+        return this.comment;
+    }
+
+    /** Returns timestamp of when comment was created. */
+    public long getTimestamp() {
+        return this.timestamp;
+    }
+  } 
+    
+    /** Method that retreives comments from the datastore and formats it. */
+    public static List<Comment> fetchComments(int commentsLimit) {
+      Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      List<Entity> results = datastore.prepare(query)
+        .asList(FetchOptions.Builder.withLimit(commentsLimit));
+      
+      List<Comment> comments = new ArrayList<>();
+      for (Entity entity : results) {
+        String comment = (String) entity.getProperty("comment");
+        long timestamp = (long) entity.getProperty("timestamp");
+        Comment newComment = new Comment(comment, timestamp);
+        comments.add(newComment);
+      }
+
+      ImmutableList<Comment> immutComments =
+        ImmutableList.<Comment>builder().addAll(comments).build(); 
+
+      return immutComments;
+    }
+
+    /** Method that adds a comment to the datastore. */
```

Comment:
please indicate that the method `throws {@link IllegalArgumentException} if the comment is empty`